dbt/include/athena/macros/materializations/snapshots/snapshot.sql

```diff
@@ -12,365 +11,112 @@
 {%- endmacro %}
 
 
-{%
+{#
     If hive table then Recreate the snapshot table from the new_snapshot_table
     If iceberg table then Update the standard snapshot merge to include the DBT_INTERNAL_SOURCE prefix in the src_cols_csv
-%}
-
-{% macro athena__snapshot_merge_sql(target, source, insert_cols, table_type) -%}
-    {% if table_type=='iceberg' %}
-        {%- set insert_cols_csv = insert_cols | join(', ') -%}
-        {%- set src_columns = [] -%}
-        {%- for col in insert_cols -%}
-          {%- do src_columns.append('dbt_internal_source.' + col) -%}
-        {%- endfor -%}
-        {%- set src_cols_csv = src_columns | join(', ') -%}
-
-        merge into {{ target }} as dbt_internal_dest
-        using {{ source }} as dbt_internal_source
-        on dbt_internal_source.dbt_scd_id = dbt_internal_dest.dbt_scd_id
-
-        when matched
-         and dbt_internal_dest.dbt_valid_to is null
-         and dbt_internal_source.dbt_change_type in ('update', 'delete')
-            then update
-            set dbt_valid_to = dbt_internal_source.dbt_valid_to
-
-        when not matched
-         and dbt_internal_source.dbt_change_type = 'insert'
-            then insert ({{ insert_cols_csv }})
-            values ({{ src_cols_csv }})
-    {%else%}
-        {%- set target_relation = adapter.get_relation(database=target.database, schema=target.schema, identifier=target.identifier) -%}
-        {%- if target_relation is not none -%}
-          {% do adapter.drop_relation(target_relation) %}
-        {%- endif -%}
-
-        {% set sql -%}
-          select * from {{ source }};
-        {%- endset -%}
-
-        {{ create_table_as(False, target_relation, sql) }}
-    {% endif %}
-{% endmacro %}
+#}
+{% macro hive_snapshot_merge_sql(target, source, insert_cols, table_type) -%}
```

Comment:
The original `athena__snapshot_merge_sql` macro is split in two separate macros, one for hive and one for iceberg. This is the hive part of that function without any other changes.